tests/kernels/test_kernels.py

```diff
@@ -395,6 +392,33 @@ def test_closest_psd_matrix(self, input, fix_diagonal, expected_output):
 
         assert np.allclose(output, expected_output, atol=1e-5)
 
+    @pytest.mark.usefixtures("skip_if_no_cvxpy_support")
+    @pytest.mark.xfail(raises=RuntimeError, reason="solver did not converge")
+    def test_closest_psd_matrix_small_perturb(self):
+        """Test obtaining the closest positive semi-definite matrix using a
+        semi-definite program with a small perturbation input.
+
+        The small perturbation ensures that the solver does not get stuck.
+
+        """
+        input, fix_diagonal, expected_output = (
+            np.array([[0, 1.000001], [1, 0]]),
+            True,
+            np.array([[1, 1], [1, 1]]),
+        )
+        try:
+            import cvxpy as cp
+
+            output = kern.closest_psd_matrix(input, fix_diagonal=fix_diagonal, feastol=1e-10)
+        except cp.error.SolverError:
```

Comment:
I think this line is a problem if cvxpy fails to import no?

Reply:
Now there's a fixture that skips the test if cvxpy is not installed.